src/status_im/i18n.cljs

```diff
@@ -52,73 +53,163 @@
+                             :zh-hant-sg  zh-hant/translations
+                             :zh-hant-hk  zh-hant/translations
+                             :zh-hant-cn  zh-hant/translations
+                             :zh-hant-mo  zh-hant/translations
+                             :zh-wuu      zh-wuu/translations
+                             :zh-yue      zh-yue/translations})
+
+;; english as source of truth
+(def trans-ids (set (keys en/translations)))
+
+(spec/def ::trans-id trans-ids)
+(spec/def ::trans-ids (spec/coll-of ::trans-id :kind set? :into #{}))
+
+(defn trans-ids-for-all-locales []
+  (->> translations-by-locale
+       (mapcat #(-> % val keys))
+       set))
+
+;; checkpoints
+
+(spec/def ::checkpoint.id keyword?)
+(spec/def ::checkpoint-defs (spec/map-of ::checkpoint.id ::trans-ids))
+
+(def checkpoint-1-trans-ids trans-ids)
+; these could be status versions instead of incremental numbers
+(def checkpoints-def (spec/assert ::checkpoint-defs
+                                  {::checkpoint|1 checkpoint-1-trans-ids
+                                ::checkpoint|2 (into checkpoint-1-trans-ids #{})}))
+(def checkpoints (set (keys checkpoints-def)))
+
+(spec/def ::checkpoint checkpoints)
+
+(def checkpoint-to-consider-locale-supported ::checkpoint|1)
+
+(defn checkpoint->trans-ids [checkpoint]
+  (get checkpoints-def checkpoint))
+
+(defn checkpoint-val-to-compare [c]
+  (-> c name (string/replace #"^.*\|" "") int))
+
+(defn >checkpoints [& cs]
+  (apply > (map checkpoint-val-to-compare cs)))
+
+;; locales
+
+(def locales (set (keys translations-by-locale)))
+
+(spec/def ::locale locales)
+(spec/def ::locales (spec/coll-of ::locale :kind set? :into #{}))
+
+(def supported-locales (spec/assert ::locales #{:en :es}))
+
+(spec/def ::supported-locale supported-locales)
+(spec/def ::supported-locales (spec/coll-of ::supported-locale :kind set? :into #{}))
+
+(defn locale->trans-ids [locale]
+  (-> translations-by-locale (get locale) keys set))
+
+(defn locale->checkpoint [locale]
+  (let [locale-trans-ids (locale->trans-ids locale)
+        checkpoint       (->> checkpoints-def
+                              (filter (fn [[checkpoint checkpoint-trans-ids]]
+                                        (set/subset? locale-trans-ids checkpoint-trans-ids)))
+                              ffirst)]
+    checkpoint))
+
+(defn locales-with-checkpoint []
+  (->> locales
+       (map (fn [locale]
+              [locale (locale->checkpoint locale)]))
+       (into {})))
+
+(defn locale-is-supported-based-on-translations? [locale]
+  (let [c (locale->checkpoint locale)]
+    (and c (or (= c checkpoint-to-consider-locale-supported)
+               (>checkpoints checkpoint-to-consider-locale-supported c)))))
+
+(defn actual-supported-locales []
+  (->> locales
+       (filter locale-is-supported-based-on-translations?)
+       set))
+
+(defn locales-with-full-support []
+  (->> locales
+       (filter (fn [l]
```

Comment:
locale instead of l

Reply:
Is this a convention as well? When it is a very small function I sometimes use smaller names since the context is very small, the same as anonymous functions use just `%`, so you get it from the small context. Will use full featured functions from now on, just for you to know why I did so. :)